compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -644,10 +688,13 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
         // alone (and this doesn't impact the SJSIR we generate).
         JSDefinitions.jsdefn.PseudoUnionType
       else
-        TypeComparer.orType(this(tp1), this(tp2), isErased = true)
+        val e1 = this(tp1)
+        val e2 = this(tp2)
+        if e1 == null || e2 == null then null
+        else TypeComparer.orType(e1, e2, isErased = true)
```

Comment:
can be streamlined by moving to `WildcardType` instead of `null`

Reply:
I switched to WildcardType but didn't find a better place to check for the special value than here, let me know if you had something specific in mind.